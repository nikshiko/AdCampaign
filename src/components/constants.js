export default {
    name: "Campaign Name",
    startDate: "Starting on",
    endDate: "Ending on",
    Budget: 'Budget'

}