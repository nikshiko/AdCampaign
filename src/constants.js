
const columnConfig = {
	name: 'Campaign name',
	startDate: 'Starting on',
	endDate: 'Ending on',
	active: 'Active',
	Budget: 'Budget',

}

export const inactive= 'Inactive'
export const active= 'Active'
export const placeholders = {
	searchText: 'Search campaign',
	startDate: 'Select start date',
	endDate: 'Select end date'
	
}

export const currency = 'USD'
export const dateFormat = 'MM/DD/YYYY'
export default columnConfig