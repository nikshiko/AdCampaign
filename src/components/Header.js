
import React from 'react'
import PropTypes from 'prop-types'
import { Input } from 'antd'

import DateRange from '../components/DateRange'
import { StyledHeader } from '../styled'
import { placeholders } from '../constants'



export default function Header ({ onChange, selectedRange }) {

	const handleSearchChange = (e) => {
		onChange({ field: 'search', value: e.target.value })
	}
	return (
		<StyledHeader>
			<DateRange
				selectedRange={selectedRange}
				handleRangeChange={onChange} />
			<Input.Search
				id="searchInput"
				placeholder={placeholders.searchText}
				onChange={handleSearchChange}
				style={{width: '200px'}} />
		</StyledHeader>
	)
}

Header.propTypes = {
	onChange: PropTypes.func.isRequired,
	selectedRange: PropTypes.shape({
		startValue: PropTypes.object,
		endValue: PropTypes.object
	})
}
