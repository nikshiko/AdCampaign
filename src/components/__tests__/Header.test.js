import React from 'react'
import Header from '../Header'
import { render, fireEvent, cleanup } from '@testing-library/react'
import { placeholders } from '../../constants'
const onChange = jest.fn().mockImplementation(()=>{})

describe('Header component',() => {
	afterEach(cleanup)
	it('should render without crashing',() => {
		const { asFragment } = render(<Header onChange={onChange} />)
		expect(asFragment()).toMatchSnapshot()
	})

	it('should trigger the onChange function when user searches for a campaign name ',() => {
		const { getByPlaceholderText } = render(<Header onChange={onChange} />)
		fireEvent.change(getByPlaceholderText(placeholders.searchText), { target: { value: 'a' } })
		expect(onChange).toHaveBeenCalledWith({'field': 'search', 'value': 'a'})
	})
})