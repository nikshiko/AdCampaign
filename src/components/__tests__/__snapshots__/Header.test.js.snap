// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa cXabmE"
  >
    <div
      class="sc-htpNat cjtfmR"
    >
      <span
        class="ant-calendar-picker"
      >
        <div>
          <input
            class="ant-calendar-picker-input ant-input"
            placeholder="Select start date"
            readonly=""
            value=""
          />
          <i
            aria-label="icon: calendar"
            class="anticon anticon-calendar ant-calendar-picker-icon"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="calendar"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
              />
            </svg>
          </i>
        </div>
      </span>
      <span
        class="ant-calendar-picker"
      >
        <div>
          <input
            class="ant-calendar-picker-input ant-input"
            placeholder="Select end date"
            readonly=""
            value=""
          />
          <i
            aria-label="icon: calendar"
            class="anticon anticon-calendar ant-calendar-picker-icon"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="calendar"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
              />
            </svg>
          </i>
        </div>
      </span>
    </div>
    <span
      class="ant-input-search ant-input-affix-wrapper"
      style="width: 200px;"
    >
      <input
        class="ant-input"
        id="searchInput"
        placeholder="Search campaign"
        type="text"
        value=""
      />
      <span
        class="ant-input-suffix"
      >
        <i
          aria-label="icon: search"
          class="anticon anticon-search ant-input-search-icon"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="search"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </i>
      </span>
    </span>
  </div>
</DocumentFragment>
`;
